src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -62,13 +65,26 @@ bool isValidElementTypeAndRank(Value val, bool donotCheckRank) {
       if (donotCheckRank)
         return true;
       // Rank must be in range of (0, 4].
-      if (!valueType.hasRank())
-        return false;
+      if (!valueType.hasRank()) {
+        std::string message = "Value does not have rank.";
+        return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+      }
       int64_t rank = valueType.getRank();
-      return ((rank > 0) && (rank <= 4));
+      if ((rank > 0) && (rank <= 4)) {
+        return true;
+      } else {
```

Comment:
Since we have `return` in the `if` part, the `else` is not necessary. For example:

```c
if ((rank > 0) && (rank <= 4))
     return true;

std::string message =
            "Rank " + std::to_string(rank) +
            " is not supported. zAIU only supports rank in range of (0, 4].";
return emitWarningMessageNNPAUnsupported(op, StringRef(message));
```

Reply:
Done. Thanks!